doc: bluetooth: mesh: update sensor docs to new API (#14234)
nrfconnect/sdk-nrf: doc/nrf/libraries/bluetooth_services/mesh/light_ctrl_cli.rst

   :local:
   :depth: 2

> .. note::
>    This model interacts with the Sensor API, which was rewritten in |NCS| v2.6.0, and this is what is documented in this release.
>    The old API is deprecated, but is still available by enabling :kconfig:option:`CONFIG_BT_MESH_SENSOR_USE_LEGACY_SENSOR_VALUE`, which is enabled by default in the deprecation period.
>    See the documentation for |NCS| v2.5.2 for documentation about the old Sensor API.
> 

Reviewer: ```suggestion
.. note::
   This model interacts with the new sensor API introduced as of |NCS| v2.6.0.
   The old API is deprecated, but still available by enabling the Kconfig option :kconfig:option:`CONFIG_BT_MESH_SENSOR_USE_LEGACY_SENSOR_VALUE`.
   The Kconfig option is enabled by default in the deprecation period.
   See the documentation for |NCS| versions prior to v2.6.0 for documentation about the old sensor API.

```

Reviewer: And the same for the other occurrences :)

Author: What to do in the other cases? The opening line is different

   :local:
   :depth: 2

> .. note::

Reviewer 2: Shouldn't this be added to the light_ctrl_srv.rst also?

Author: Should it? As far as I remember there is no API changes in the server, only changes to internal implementation details. Basically, if there is no difference in the 2.5 and 2.6 docs (which I believe is the case for light_ctrl_srv?) there is no point in having this note about looking up stuff in the old docs.

>    This model interacts with the new sensor API introduced as of |NCS| v2.6.0.

Reviewer 2: ```suggestion
   A new sensor API is introduced as of |NCS| v2.6.0.
```
Same as in the other files?

   :local:
   :depth: 2

.. note::
   This model interacts with the new sensor API introduced as of |NCS| v2.6.0.

>    As a consequence, parts of the model API have been changed as well.

Author: @mia-ko I added this line, please check it
